src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1595,6 +1598,33 @@ namespace winrt::TerminalApp::implementation
                 }
             }
         }
+        else if (executableFilename == L"pwsh" || executableFilename == L"pwsh.exe")
+        {
+            // is does executablePath start with %ProgramFiles%\\PowerShell?
+            const std::filesystem::path powershellCoreRoot
+            {
+                wil::ExpandEnvironmentStringsW<std::wstring>(
+#if defined(_M_AMD64) || defined(_M_ARM64) // No point in looking for WOW if we're not somewhere it exists
+                    L"%ProgramFiles(x86)%\\PowerShell"
+#elif defined(_M_ARM64) // same with ARM
+                    L"%ProgramFiles(Arm)%\\PowerShell"
+#else
+                    L"%ProgramFiles%\\PowerShell"
```

Comment:
Wait, I'm a bit confused. On amd64, we check whether it is only under `Program Files (x86)`?

Reply:
Honestly, I was confused too. I thought I copied the logic from https://github.com/microsoft/terminal/blob/7db7ba1ac9c1ee65f7b389f1893f985f10e68c17/src/cascadia/TerminalSettingsModel/PowershellCoreProfileGenerator.cpp#L264-L278

and it seems to work, but it doesn't make sense to me. Some sort of ProgramFiles virtualization or something?

Reply:
So, that code calls `_accumulate...Directory` like this:

```
Program Files       ──┐ ──┐ ──┐
                      │   │   │
Program Files (x86)   │ ──┤   │
                      │   │   │
Program Files (Arm) ──┤   │   │
                      │   │   └─ Runs on x86 only
                      │   │
                      │   └───── Runs on x86 and x64
                      │
                      └───────── Runs on arm64, x86 and x64
```

So: arm64 has the possibility of hitting all 3 holders; x64 only checks the top two (since Program Files Arm doesn't exist there) and x86 only checks the top (since x86 doesn't have its own Program Files x86 folder)